src/custom/utils/metadata.ts

```diff
@@ -0,0 +1,38 @@
+interface IMetadata {
+  kind: MetadataKind
+  version: string
+}
+
+export enum MetadataKind {
+  REFERRAL = 'referrer',
+}
+
+export type Metadata = {
+  [MetadataKind.REFERRAL]: IReferralMetadata
+}
+
+export type MetadataDoc = {
+  version: string
+  appCode?: string
+  metadata: Metadata
+}
+
+export interface IReferralMetadata extends IMetadata {
+  referrer: string
+}
+
+export const DEFAULT_APP_CODE = 'CowSwap'
+
+export const generateReferralMetadataDoc = (referralAddress: string): MetadataDoc => {
```

Comment:
Nitpick: My personal preference is to write regular functions unless there's a benefit into using const based functions.

Also up to you.

But if you decide to stick to const based function, you might as well abbreviate it a little and swallow the return statement

Reply:
Here I also go with the flow, whatever the team prefers